internal/daemonset/daemonset.go

```diff
@@ -0,0 +1,194 @@
+	"github.com/test-network-function/cnf-certification-test/pkg/tnf"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	daemonSetName    = "partner-repo"
+	namespace        = "default"
+	containerName    = "container-00"
+	imageWithVersion = "quay.io/testnetworkfunction/debug-partner:latest"
+	timeout          = 5 * time.Minute
+	nodeExporter     = "node-exporter"
+)
+
+//nolint:funlen
+func CreateDaemonSetsTemplate(dsName, namespace, containerName, imageWithVersion string) *v1.DaemonSet {
+	dsAnnotations := make(map[string]string)
+	dsAnnotations["debug.openshift.io/source-container"] = containerName
+	dsAnnotations["openshift.io/scc"] = nodeExporter
+	matchLabels := make(map[string]string)
+	matchLabels["name"] = dsName
+
+	var trueBool = true
+	var zeroInt int64
+	var zeroInt32 int32
+	var preempt = corev1.PreemptLowerPriority
+	var tolerationsSeconds int64 = 300
+	var hostPathType = corev1.HostPathDirectory
+
+	container := corev1.Container{
+		Name:            containerName,
+		Image:           imageWithVersion,
+		ImagePullPolicy: "IfNotPresent",
+		SecurityContext: &corev1.SecurityContext{
+			Privileged: &trueBool,
+			RunAsUser:  &zeroInt,
+		},
+		Stdin:                  true,
+		StdinOnce:              true,
+		TerminationMessagePath: "/dev/termination-log",
+		TTY:                    true,
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				MountPath: "/host",
+				Name:      "host",
+			},
+		},
+	}
+	return &v1.DaemonSet{
+
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        dsName,
+			Namespace:   namespace,
+			Annotations: dsAnnotations,
+		},
+		Spec: v1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: matchLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: matchLabels,
+				},
+				Spec: corev1.PodSpec{
+					Containers:       []corev1.Container{container},
+					PreemptionPolicy: &preempt,
+					Priority:         &zeroInt32,
+					HostNetwork:      true,
+					Tolerations: []corev1.Toleration{
+						{
+							Effect:            "NoExecute",
+							Key:               "node.kubernetes.io/not-ready",
+							Operator:          "Exists",
+							TolerationSeconds: &tolerationsSeconds,
+						},
+						{
+							Effect:            "NoExecute",
+							Key:               "node.kubernetes.io/unreachable",
+							Operator:          "Exists",
+							TolerationSeconds: &tolerationsSeconds,
+						},
+						{
+							Effect: "NoSchedule",
+							Key:    "node-role.kubernetes.io/master",
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "host",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/",
+									Type: &hostPathType,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+func DeleteDaemonSet(daemonSetName, namespace string) error {
```

Comment:
Please add a comment before this function as it is to be exported.

Reply:
you're right. I did it.